### PR TITLE
Tracking PR for release `v0.31.0-rc1`

### PR DESCRIPTION
**Pull request by tcharding**
Lets go!

As discussed in #2099 we are going to do RC releases for this release. This PR is for the first one, as such it does not include changelog entries, we are going to try out using the new github action to create those.

### CODE REVIEW

**Comment by vincenzopalazzo** [comment]
I do not think if this is taken into account, but there is any change that this will go in? https://github.com/rust-bitcoin/rust-bitcoin/pull/2107 and maybe https://github.com/rust-bitcoin/rust-bitcoin/pull/2090?

**Comment by apoelstra** [comment]
On the one hand, I think we should try to slip the POW stuff in because it's currently broken.

But on the other hand I think we should try harder to stick to a regular release cadence (and yes, I appreciate that this release was supposed to happen at the end of September, but we've had inconsistent availability of maintainers) so that we don't feel so much pressure to "slip stuff in".

Maybe we should ship the RC without it, and try to get the PoW stuff into the next RC in a week? This isn't really in the spirit of an RC, which should be feature-complete, but I really don't want to delay this release further.

**Comment by vincenzopalazzo** [comment]
Concept ACK @apoelstra 

I am happy also with just the target bug fix :) because this is really bad on nakamoto client

**Comment by tcharding** [comment]
The pow stuff is uncovering a bunch of things, I'll work on it today push PRs with an aim to keep bug fixes separate so they can go in the next rc and improvements can be left for next release.

**Review by apoelstra — APPROVED** [review]
ACK 6731e85dc1ee7e516211efaed51cf14f21078ee2

**Comment by Kixunil** [comment]
Are we waiting on anything? This has a weak ack from me: I don't have any specific objection to this but I am unable to judge the changes made while I was away to be sure there are no significant problems. I have to assume reviews went fine. If anyone wants me to look at anything specific I'll be happy to.

**Comment by apoelstra** [comment]
@Kixunil it needs a second ack from a maintainer. If you are willing to write "ACK <commit ID>" that's enough, even if it's a "weak ACK". There isn't much to review here, just "should we push a review out now or not".

A@sanket1729 is away this week I think. Or @clarkmoody has been active recently and might be available.

**Review by Kixunil — APPROVED** [review]
ACK 6731e85dc1ee7e516211efaed51cf14f21078ee2

**Comment by apoelstra** [comment]
Tagged and published.

**Comment by apoelstra** [comment]
I dunno wtf [this is supposed to be](https://github.com/rust-bitcoin/rust-bitcoin/releases/tag/bitcoin-0.30.2) but I think we can call our automated release thing a total failure.

**Comment by tcharding** [comment]
TL;DR I'm ok to drop the tool experiment and just do it manually.

FTR I think its github not the github action setup we have. I believe github's release tooling thing cannot handle the tag name we are using. I went to https://github.com/rust-bitcoin/rust-bitcoin/releases/new and had a play generating various release notes and it works for the 0.29.0 and 0.29.2 releases but not for ones that start with a word (`bitcoin-0.30.0`). I don't want to spend anymore time playing with it right now. I'll hack up release notes for 0.31.0 manually, perhaps we can re-visit this during the next dev cycle?

**Comment by apoelstra** [comment]
> I'll hack up release notes for 0.31.0 manually, perhaps we can re-visit this during the next dev cycle?

Thank you!! And yes, this sounds great.

**Comment by Kixunil** [comment]
We won't suffer with changelogs anymore (I hope :D) - [mkchlog now exists](https://github.com/crywolf/mkchlog)! There are still a few things to improve but it can at least properly check commits and produce something useful.

**Comment by tcharding** [comment]
That crate looks really interesting, how ready is it? Should we consider giving it a go during the next dev cycle?

**Comment by Kixunil** [comment]
I think we could try soon, I would just wait for my PRs to get in (especially `#3`). From what I know `mkchlog check` already works correctly and if there's any problems with generating we can fix them later/manually.
